utils/db-parts.stanza

```diff
@@ -160,54 +165,67 @@ defmethod to-jitx (resistor: Resistor) -> Instantiable :
+public defn Resistor (user-properties:Tuple<KeyValue>, exist:Tuple<String>) -> Resistor :
+  Resistor(user-properties, exist, OPTIMIZE-FOR, OPERATING-TEMPERATURE)
 
-public defn Resistor (properties:Tuple<KeyValue>, exist:Tuple<String>, sort:Tuple<String>, temperature-properties:Tuple<KeyValue>) -> Resistor :
+public defn Resistor (user-properties:Tuple<KeyValue>, exist:Tuple<String>, sort:Tuple<String>, operating-temperature:[Double, Double]|False) -> Resistor :
+  val query-properties = resistor-query-properties(user-properties, exist, sort, operating-temperature)
   ;Query the database with the given properties
-  val query-properties = to-tuple $ cat-all([["category" => "resistor",
-                                              "_sort" => sort,
-                                              "_exist" => exist],
-                                             temperature-properties,
-                                             properties])
-
-  val json = dbquery-first(query-properties) as JObject
-  Resistor(json)
+  Resistor $ dbquery-first(query-properties) as JObject
   ; Resistor("manufacturer", "mpn", "trust", 0.0, 0.0, 0.0, "mounting", false, false, Sourcing(false, 0, 0), [], "type", false, 0.0, false, false, false)
 
-public defn Resistors (properties:Tuple<KeyValue>, exist:Tuple<String>) -> Tuple<Resistor> :
-  val sort = OPTIMIZE-FOR
-  val temperature-properties = ["max-rated-temperature.min" => OPERATING-TEMPERATURE[0],
-                                "min-rated-temperature.max" => OPERATING-TEMPERATURE[1]]
+public defn Resistors (user-properties:Tuple<KeyValue>, exist:Tuple<String>) -> Tuple<Resistor> :
+  val query-properties = resistor-query-properties(user-properties, exist, OPTIMIZE-FOR, OPERATING-TEMPERATURE)
+  ;Query the database with the given properties
+  map{Resistor, _} $ dbquery-all(query-properties) as Tuple<JObject>
+
+defn resistor-query-properties (user-properties:Tuple<KeyValue>,
+                                exist:Tuple<String>,
+                                sort:Tuple<String>,
+                                operating-temperature:[Double, Double]|False) -> Tuple<KeyValue> :
+  query-properties(user-properties,
+                   exist,
+                   sort,
+                   operating-temperature,
+                   "resistor",
+                   false)
+
+;============================================================
+;======================== Look-ups ==========================
+;============================================================
 
-  val query-properties = to-tuple $ cat-all([["category" => "resistor",
-                                              "_sort" => sort,
-                                              "_exist" => exist],
-                                             temperature-properties,
-                                             properties])
+public defn look-up-resistors (attribute: String) -> Tuple :
```

Comment:
What is the return type, `Tuple<___>`?

Reply:
It depends

Reply:
Yes getting a `JObject` is not great, in this example it would be better to have a `MinMaxRange` but that would be more work